api/errors/error.go

```diff
@@ -0,0 +1,25 @@
+package errors
+
+// This file contains all of the errors that can be generated from the
+// docker engine but are not tied to any specific top-level component.
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+const errGroup = "engine"
+
+var (
+	// ErrorCodeGeneric is generated when there is no better error message
+	// available. This will typically happen when some low-level function
+	// is called and we just need to bubble the unknown error back up to
+	// the API. We should try to avoid using this.
+	ErrorCodeGeneric = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
The error message associated with that error code isn't appropriate for this engine situation - I wanted something more specific to the engine - and to say something about the logs, which the errorcode package probably can't assume in general.